onyo/commands/new.py

```diff
@@ -22,6 +22,7 @@ def create_assets_in_destination(assets: dict, repo: Repo) -> None:
             asset.touch()
         repo._write_asset(asset, assets[asset])
     repo.add(list(assets.keys()))
+    repo.clear_caches()
```

Comment:
IMO, this should be made part of `repo.add()`, so that it always does the right thing internally, rather than requiring consumers to maintain the state of non-public structures internal to `repo`.

Also, I think it would be better if it was more granular which caches are cleared, rather than always all.

Reply:
I added it to `Repo.add()`, though did not modify the cache-clearing behavior for more granularity, since I think it is not clear to me, that we reliably know, if assets, directories, templates or else were changed.
This will be even more true, as soon as we add functionality to easier modify e.g. templates.

Reply:
I want to ping you on this one, @aqw because of what you wrote in the downstairs review about this being the most controversial part😇

I thought about it, too, and think from an API perspective it makes most sense to do this in the repo class, since "add" almost literally is saying to the repo "something changed".